node/core/dispute-coordinator/src/real/initialized.rs

```diff
@@ -533,31 +588,45 @@ impl Initialized {
 				}
 			},
 			DisputeCoordinatorMessage::RecentDisputes(tx) => {
-				let recent_disputes = overlay_db.load_recent_disputes()?.unwrap_or_default();
-				let _ = tx.send(recent_disputes.keys().cloned().collect());
+				// Return error if session information is missing.
+				if let Some(subsystem_error) = self.error.clone() {
+					return Err(Error::NonFatal(NonFatal::RollingSessionWindow(subsystem_error)))
+				} else {
+					let recent_disputes = overlay_db.load_recent_disputes()?.unwrap_or_default();
+					let _ = tx.send(recent_disputes.keys().cloned().collect());
+				}
 			},
 			DisputeCoordinatorMessage::ActiveDisputes(tx) => {
-				let recent_disputes =
-					overlay_db.load_recent_disputes()?.unwrap_or_default().into_iter();
-				let _ =
-					tx.send(get_active_with_status(recent_disputes, now).map(|(k, _)| k).collect());
+				if let Some(subsystem_error) = self.error.clone() {
+					return Err(Error::NonFatal(NonFatal::RollingSessionWindow(subsystem_error)))
+				} else {
+					let recent_disputes =
+						overlay_db.load_recent_disputes()?.unwrap_or_default().into_iter();
```

Comment:
nit: This is very implicit, and `if let Some(disputes) = x { load_recent_disputes(disputes, now) } else { Vec::new() }` reduces visual complexity a bit